scrooge-benchmark/src/main/scala/com/twitter/scrooge/benchmark/TByteArrayOutputStream.scala

```diff
@@ -0,0 +1,41 @@
+package com.twitter.scrooge.benchmark
+
+import com.twitter.scrooge.TUnboundedByteArrayOutputStream
+import java.util.concurrent.TimeUnit
+import org.apache.thrift.TByteArrayOutputStream
+import org.openjdk.jmh.annotations._
+
+object TByteArrayOutputStreamBenchmark {
+  @State(Scope.Thread)
+  class TByteArrayOutputStreamState {
+    @Param(Array("1000", "10000"))
+    var times: Int = 1000
+
+    @Param(Array("512", "1024"))
+    var initialSize: Int = 512
+
+    val bytes: Int = 16
+
+    def run(out: TByteArrayOutputStream) {
+      val buf = new Array[Byte](bytes)
+      for (i <- 1 to times) {
+        out.write(buf, 0, bytes)
+        out.reset()
+      }
+    }
+  }
+}
+
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Array(Mode.Throughput))
+class TByteArrayOutputStreamBenchmark {
+  import TByteArrayOutputStreamBenchmark._
+
+  @Benchmark
+  def timeTByteArrayOutputStream(state: TByteArrayOutputStreamState) =
+    state.run(new TByteArrayOutputStream(state.initialSize))
```

Comment:
JMH has a very opinionated style of writing code and this goes against that.

instead, what you can do:
* lift the output streams into your `@State` class. then "inject" a state instance into your `Benchmarks`.  as an example, look at https://github.com/twitter/finagle/blob/develop/finagle-benchmark/src/main/scala/com/twitter/finagle/stats/StatsReceiverBenchmark.scala
* lift the creation of the `Array[Byte]` into the `State` as you want to remove that work from the test.
* You do not want a loop in the `run` method. it should only call `write` and `reset`. JMH will "do the looping".